src/rgw/rgw_op.cc

```diff
@@ -5232,6 +5232,57 @@ int RGWCopyObj::verify_permission(optional_yield y)
 					      RGW_PERM_READ)) { 
 	  return -EACCES;
 	}
+
+  // Check for tagging permissions
+  auto tagging_drctv = s->info.env->get("HTTP_X_AMZ_TAGGING_DIRECTIVE");
+  int itag_der;
+  if (tagging_drctv) {
+    if (strcasecmp(tagging_drctv, "COPY") == 0)
```

Comment:
use strncasecmp, its using user supplied value, should be stricter check

Reply:
used string compare function.